content/usgs-packages/geoknife_Discovery.Rmd

```diff
@@ -0,0 +1,233 @@
+
+## Available webgeoms
+
+The next component to `geonkife` jobs is the spatial extent of the data, a.k.a. the stencil. The stencil is defined by using either of the functions `simplegeom` or `webgeom`. `simplegeom` is used to explicitly define an area by the user, but `webgeom` is used to specify an existing web feature service (WFS) as the geospatial extent. Defining your stencil using `simplegeom` will be covered in the next lesson. This lesson will just show how to learn what available webgeoms exist. Users can use any WFS url to create their stencil, but there are a number of features that exist through GDP already. To determine what features exist, you can create a default webgeom object and then query its geom name, attributes, and values. This will return all available GDP default features.
+
+```{r create-stencil, message=FALSE, warning=FALSE}
+# setup a default stencil by using webgeom and not supplying any arguments
+default_stencil <- webgeom()
+
+# now determine what geoms are available with the default
+default_geoms <- query(default_stencil, "geoms")
+length(default_geoms)
+head(default_geoms)
+```
+
+You will notice a pattern with the names of the geoms: a category followed by `:`, and then a specific name. See the table below for definitions of the categories. These category-name combinations are the strings you would use to define your geom.
+
+```{r cateogry-table, eval=TRUE, echo=FALSE, message=FALSE}
+Category <- c("upload", "sample", "draw", "derivative")
+Description <- c(
+  "upload",
+  "sample",
+  "draw",
+  "derivative"
+)
+
+library(htmlTable)
+htmlTable(data.frame(Category, Description),
+          rnames=FALSE, align=c("l","l"), col.rgroup = c("none", "#F7F7F7"), 
+          css.cell="padding-bottom: 0.5em; padding-right: 0.5em; padding-top: 0.5em;")
+```
+
+Similar to fabrics where you could not query times without setting the variables, you cannot query attributes of stencils before defining the geoms. Likewise, you cannot query for values of a stencil until you have set the attributes. Attributes give the metadata associated with the stencil and it's geom. Values tell you the individual spatial features available in that attribute of the geom. 
+
+```{r explore-stencil, message=FALSE, warning=FALSE}
+# add a geom to see what values are available
+geom(default_stencil)
+geom(default_stencil) <- "sample:CONUS_states"
+
+# now that geom is set, you can query for available attributes
+query(default_stencil, "attributes")
+attribute(default_stencil) <- "STATE"
+
+# now that attribute is set, you can query for available values
+STATE_values <- query(default_stencil, "values")
+head(STATE_values)
+
+# switch the stencil to see the differences
+ecoreg_stencil <- default_stencil
+geom(ecoreg_stencil) <- "sample:Ecoregions_Level_III"
+query(ecoreg_stencil, "attributes")
+attribute(ecoreg_stencil) <- "LEVEL3_NAM"
+ecoreg_values <- query(ecoreg_stencil, "values")
+head(ecoreg_values)
```

Comment:
Below you use the shortcut for setting values for the webgeom, but it is probably good to add a `values(ecoreg_stencil) <- ...` here to show the way to set that in the stencil.